[change] Documentation update

beanpole135 opened the pull request:
Large documentation update of all the build options for the main "make" command, as well as an updated build manifest readme with all the supported fields/options for version 1.1 of the manifest system.

Also fix an oversight in the local source dir path: Allow it to be defined by an environment variable and only use the "source" relative path as the default value if the env variable is not defined (no change to default behavior).

beanpole135 committed:
Large documentation update for the build system.
Still in-progress, but most things seem to be listed now. Still need to verify some details such as output directories and such.

beanpole135 committed:
Update readme a bit more with output directory structure.

beanpole135 committed:
Last minor updates to readme.

beanpole135 committed:
Start updating the manifests Readme.

beanpole135 committed:
More cleanup of the build manifest readme.
Start removing options that are no longer valid with v1.1 of manifest, and add warnings to some options that remain.

beanpole135 committed:
Remove a couple more options for the old base packages framework

darkfiberiru commented on README.md:
```diff
@@ -8,14 +8,75 @@ TrueOS Build repo is a JSON manifest-based build system for TrueOS. It uses poud
+
+### make config
+This will launch an interactive prompt to select a build manifest from the example files in the [manifests directory](https://github.com/trueos/build/tree/master/manifests) and use that as the default build manifest.
+This selection will get saved to the local ".config/manifest" file and used whenever the **TRUEOS_MANIFEST** environment variable is not set. This command may be run whenever a change your default manifest is desired.
```
```suggestion
The selected Manifest name will get saved to the local ".config/manifest" file and used whenever the **TRUEOS_MANIFEST** environment variable is not set. This command may be run whenever a different default manifest is desired.
```

darkfiberiru commented on README.md:
```diff
@@ -8,14 +8,75 @@ TrueOS Build repo is a JSON manifest-based build system for TrueOS. It uses poud
+**WARNING:** If no build manifest is specified (either by running `make config` or providing the TRUEOS_MANIFEST environment variable), then the "trueos-snapshot.json" build manifest will be used automatically from the [manifests directory](https://github.com/trueos/build/tree/master/manifests).
+
+### make ports
+Build all of the OS and ports and assemble a package repository.
```
```suggestion
This will build the OS and ports followed by assembling a package repository.
```


Ok Don't love second half of my suggestion but feel somethings needed here.

beanpole135 replied:
I just adjusted the phrasing in that section a bit more.

beanpole135 committed:
Quick update. Mention that the auto-install-packages option is used when assembling a VM image.

darkfiberiru committed:
Update README.md

Co-Authored-By: beanpole135 <ken@ixsystems.com>

beanpole135 commented:
Just FYI: Don't merge this just yet. Still auditing all the build manifest changes with v1.1 right now.

beanpole135 committed:
Add the new "vm" object to the build manifest documentation

beanpole135 committed:
Add support for a vm.auto-install-packages list in addition to the iso.auto-install-packages list.

beanpole135 commented:
Ok, I finished my first pass through all the scripts that use manifest options.
I probably still missed a few things though, so I would strongly recommend a detailed review.

beanpole135 committed:
A few wording changes.

kmoore134 commented on README.md:
```diff
@@ -8,14 +8,79 @@ TrueOS Build repo is a JSON manifest-based build system for TrueOS. It uses poud
+This will launch an interactive prompt to select a build manifest from the example files in the [manifests directory](https://github.com/trueos/build/tree/master/manifests) and use that as the default build manifest.
+The selected Manifest name will get saved to the local ".config/manifest" file and used whenever the **TRUEOS_MANIFEST** environment variable is not set. This command may be run whenever a different default manifest is desired.
+
+**WARNING:** If no build manifest is specified (either by running `make config` or providing the TRUEOS_MANIFEST environment variable), then the "trueos-snapshot.json" build manifest will be used automatically from the [manifests directory](https://github.com/trueos/build/tree/master/manifests).
```
Its trueos-snapshot-builder.json now.

beanpole135 replied:
Fixed.
Apparently the build.sh script was never updated to look for the "-builder" manifest by default either - so I went ahead and fixes that in build.sh as well.

beanpole135 committed:
Chase the change of the default manifest file.
1. Update docs (trueos-snapshot-builder.json)
2. Fix build.sh so that it looks for trueos-snapshot-builder.json instead of just trueos-snapshot.json.
